components/FA2Display.tsx

```diff
@@ -0,0 +1,76 @@
+import { fold } from "fp-ts/Either";
+import { pipe } from "fp-ts/function";
+import * as t from "io-ts";
+import React from "react";
+import Alias from "./Alias";
+
+const FA2DataCodec = t.array(
+  t.type({
+    fa2_address: t.string,
+    name: t.string,
+    token_id: t.number,
+    to: t.string,
+    imageUri: t.union([t.string, t.undefined]),
+    amount: t.string,
+  })
+);
+
+const FA2Display: React.FC<{ data: string }> = ({ data }) => {
+  let jsonData;
+
+  try {
+    jsonData = JSON.parse(data);
+  } catch (error) {
+    console.error("Error parsing JSON:", error);
+    return <div>{data}</div>;
+  }
```

Comment:
This should be refactored with a useEffect. If we do it this way, it means that on every render we go through this `try/catch` and it's quite heavy

Reply:
Also, after a second thought, I don't think this should be here, we should provide to the component the data it needs to render. 
Imo component's job is rendering UI, not doing parsing, so it shouldn't be done here